### PR TITLE
add Knowledge Distillation

### PR DESCRIPTION
**Pull request by ifromeast**

### CODE REVIEW

**Comment by hijkzzz** [comment]
Reference Implemenration: https://github.com/microsoft/LMOps/tree/main/minillm

**Comment by catqaq** [comment]
test results: for 7b-chat distill 7b base
alpaca_data.json: 1h50mins
alpaca_data_1k: 3mins

**Comment by catqaq** [comment]
> test results: for 7b-chat distill 7b base alpaca_data.json: 1h50mins alpaca_data_1k: 3mins

thx, it seems works well.I will merge it first.
Welcome to conduct more scale testing and performance testing.

File: openrlhf/trainer/kd_trainer.py
Comment (catqaq) [inline openrlhf/trainer/kd_trainer.py:1]: For later optimization/expansion, here can use a function like get_teacher_knowledge.
` with torch.no_grad():
                    teacher_logits = self.teacher_model(inputs, attention_mask=attention_mask, return_output=True)[
                        "logits"
                    ]`

**Review by catqaq — APPROVED** [review]
